index.html

```diff
@@ -2532,183 +2532,169 @@ <h1>Servient Implementation</h1>
-                local hardware and devices behind proprietary
-                communications protocols may also represented as <a>Things</a>
-                in the runtime environment, that is, they can be made
-                accessible via <a>Consumed Thing</a> abstractions. The <a>WoT
-                    Runtime</a> is also tasked with generating and exposing a <a>Thing
-                    Description</a> based on general metadata (e.g.,
-                location), application metadata (e.g., <a>Interaction
-                    Affordance</a>), and communication metadata for the
-                available <a>Protocol Bindings</a>
-                (e.g., based on <a>WoT Binding Templates</a>).
+            <p>
+                See [[?wot-scripting-api]] for more details.
             </p>
         </section>
         <section>
-            <h2>Scripting Runtime</h2>
-            <p> When the <a>WoT Scripting API</a> is used an additional
-                nested <a>Scripting Runtime</a> is needed, which is a
-                language runtime to actually manage and execute scripts.
-                The <a>Scripting Runtime</a> runs inside of but is
-                distinct from the <a>WoT Runtime</a>.
+            <h2>Exposed Thing and Consumed Thing Abstractions</h2>
+            <p>
+                The <a>WoT Runtime</a> instantiates software representations of <a>Things</a> based on their <a>TDs</a>.
+                These software representations provide the interface toward the behavior implementation.
+            </p>
+            <p>
+                The <a>Exposed Thing</a> abstraction represents a <a>Thing</a> hosted locally 
+                and accessible from the outside through the protocol stack implementation of the <a>Servient</a>.
+                The behavior implementation can fully control <a>Exposed Things</a> by defining their metadata and <a>Interaction Affordances</a>,
+                and providing their autonomous behavior.
+            </p>
+            <p>
+                The <a>Consumed Thing</a> abstraction represents a remotely hosted <a>Thing</a> for <a>Consumers</a> that needs to be accessed using a communication protocol.
+                <a>Consumed Things</a> are proxy objects or stubs.
+                The behavior implementation is restricted to reading their metadata and activating their <a>Interaction Affordances</a> as described in the corresponding <a>TD</a>.
+                <a>Consumed Things</a> can also represent system features such as local hardware or devices behind proprietary or legacy communication protocols.
+                In this case, the <a>WoT Runtime</a> must provide the necessary adaptation between system API and <a>Consumed Thing</a>.
+                Furthermore, it must provide corresponding <a>TDs</a> and make them available to the behavior implementation,
+                for instance, by extending a discovery mechanism.
```

Comment:
What does this sentence mean? How do you extend a discovery mechanism?

Reply:
Added reference to the WoT Scripting API discover() method as illustration.